src/app/product-details-page/components/product-description/product-image-preview/zoomit/zoomit.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, OnInit, Input, OnChanges } from '@angular/core';
+
+@Component({
+  selector: 'app-zoomit',
+  templateUrl: './zoomit.component.html',
+  styleUrls: ['./zoomit.component.scss']
+})
+export class ZoomitComponent implements OnInit, OnChanges {
+  // @Input () zsrc: string;
+  zsrc = 'http://www.therxreview.com/wp-content/uploads/2013/05/Reebok-Track-Jacket.png';
+  public pos: string;
+  public zUrl: string;
+  isHovered = false;
+
+  ngOnInit() {
+  this.zUrl = `url(${this.zsrc})`;
+  this.pos = '300% 300%';
```

Comment:
the default values should be defined above. (ngOnInit should be used for ajax calls, running some logic on start etc, subscription on event etc...)

Reply:
move the default position to css